app/server/appsmith-server/src/main/java/com/appsmith/server/services/CommentServiceImpl.java

```diff
@@ -222,7 +222,15 @@ public CommentServiceImpl(
             } else {
                 return publishEmail.thenReturn(savedComment);
             }
-        });
+        }).flatMap(
+                createdComment -> {
+                    Map<String, Object> extraProperties = new HashMap<>();
+                    if(CommentUtils.isAnyoneMentioned(createdComment)) {
+                        extraProperties.put("tagged", true);
+                    }
+                    return analyticsService.sendCreateEvent(createdComment, extraProperties);
+                }
+        );
```

Comment:
In this implementation, we are sending the `tagged` property, only if it is true. When no one is tagged, we aren't sending the `tagged` property at all. Can we send it as `false` when no one is tagged? Having properties be available in the extra properties always makes it a lot easier in handling them in our downstream pipelines. Something like this:

```suggestion
        })
        .flatMap(createdComment ->
                analyticsService.sendCreateEvent(createdComment, Map.of("tagged", CommentUtils.isAnyoneMentioned(createdComment)))
        );
```